tm2py/components/network/transit/transit_assign.py

```diff
@@ -452,14 +452,22 @@ def run(self):
             self.transit_network.update_auto_times(time_period)
 
             if self.controller.iteration == 0:
+                # iteration = 0 : run uncongested transit assignment
                 use_ccr = False
                 congested_transit_assignment = False
                 print("running uncongested transit assignment with warmstart demand")
                 self.run_transit_assign(
                     time_period, use_ccr, congested_transit_assignment
                 )
-
-            else:  # iteration >=1
+            elif (self.controller.iteration == 1) & (self.controller.config.warmstart.use_warmstart_skim):
+                # iteration = 1 and use_warmstart_skim = True : run uncongested transit assignment
```

Comment:
Why so? is it to compromise for the fact that no warmstart transit assignment was run in global iteration 0, or is it to save run time?

Reply:
Yes, because when using the warm start with the skim option, we skip the transit skim and assignment in the global iteration 0. In iteration 1, the generated transit demand for an OD pair could be non-zero, even if the skim for that OD pair is zero in the warm start skim file (making it inconsistent), and thereby crashing in the congested transit assignment.

I ran into this above issue before making the code change.